examples/js/loaders/GLTFLoader.js

```diff
@@ -2772,126 +2831,148 @@ THREE.GLTFLoader = ( function () {
 
 		var animationDef = json.animations[ animationIndex ];
 
-		return this.getMultiDependencies( [
+		var nodePendings = [];
+		var inputAccessorPendings = [];
+		var outputAccessorPendings = [];
+		var samplerPendings = [];
+		var targetPendings = [];
 
-			'accessor',
-			'node'
+		for ( var i = 0, il = animationDef.channels.length; i < il; i ++ ) {
 
-		] ).then( function ( dependencies ) {
+			var channel = animationDef.channels[ i ];
+			var sampler = animationDef.samplers[ channel.sampler ];
 
-			var tracks = [];
+			if ( sampler === undefined ) continue;
 
-			for ( var i = 0, il = animationDef.channels.length; i < il; i ++ ) {
+			var target = channel.target;
+			var name = target.node !== undefined ? target.node : target.id; // NOTE: target.id is deprecated.
+			var input = animationDef.parameters !== undefined ? animationDef.parameters[ sampler.input ] : sampler.input;
+			var output = animationDef.parameters !== undefined ? animationDef.parameters[ sampler.output ] : sampler.output;
 
-				var channel = animationDef.channels[ i ];
-				var sampler = animationDef.samplers[ channel.sampler ];
+			nodePendings.push( this.getDependency( 'node', name ) );
+			inputAccessorPendings.push( this.getDependency( 'accessor', input ) );
+			outputAccessorPendings.push( this.getDependency( 'accessor', output ) );
+			samplerPendings.push( sampler );
+			targetPendings.push( target );
```

Comment:
I'm OK with the way you've written this (it is a hard case...) but just to suggest one more concise alternative:

```js
var animationData = Promise.all( [

  this.getDependency( 'node', name ),
  this.getDependency( 'accessor', input ),
  this.getDependency( 'accessor', output )

] ).then( function ( deps ) {
  
  animationData.push( {
    node: deps[ 0 ],
    input: deps[ 1 ],
    output: deps[ 2 ],
    sampler: sampler,
    target: target
  } );

} );

pending.push( animationData );

...

Promise.all( pending ).then( function () {
  
  for ( var i = 0; il = animationData.length; i < il; i ++ ) { ... }

} );

Reply:
I first tried that but realized elements of `animationData` array can be out of order, that leads to out of order elements of `tracks`. It won't be problem so much, but fixed order would be less problem. Actually I've had experience I wrote like `editTrack( animation.tracks[ 129 /* hard coded */ ] )` for specific glTF model for some experiments.